lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -1872,6 +1872,12 @@ ThreadContext::PushEntryExitRecord(Js::ScriptEntryExitRecord * record)
         Assert(lastRecord->leaveForHost || lastRecord->leaveForAsyncHostOperation);
         lastRecord->hasReentered = true;
         record->next = lastRecord;
+
+        // these are on stack, which grows down. if this condition doesn't hold, then the list somehow got messed up
+        if ((uintptr_t)record > (uintptr_t)lastRecord)
+        {
+            RaiseFailFastException(NULL, NULL, NULL);
```

Comment:
I think in our codebase we generally use NULL rather than nullptr when calling Windows APIs. Dumps from FailFast exception should contain same information as any other crash, so I think there's no need to record anything special here.